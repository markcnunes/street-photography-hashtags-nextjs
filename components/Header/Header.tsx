import styled from '@emotion/styled';
import Link from 'next/link';
import Navbar from 'components/NavBar';
import Button from 'components/Button';

const StyledHeader = styled('header')((props) => ({
  display: 'flex',
  justifyContent: 'space-between',
  flexWrap: 'wrap',
  alignItems: 'center',
  width: '100%',
  padding: 20,
  color: props.theme.colors.white,
  '.logo': {
    display: 'flex',
    alignItems: 'center',
    marginRight: 10,
    img: {
      width: 100,
    },
  },
  '.github': {
    marginRight: 5,
    width: 20,
    height: 20,
  },
  '@media (max-width: 400px)': {
    justifyContent: 'center',
  },
  '@media (max-width: 780px)': {
    nav: {
      order: 1,
      width: '100%',
      justifyContent: 'center',
    },
  },
}));

const Header: React.FC = () => {
  return (
    <StyledHeader>
      <Link href='/'>
        <a className='logo' title='home'>
          <img src={'./logo-white.svg'} alt='Logo' />
        </a>
      </Link>

      <Navbar />
      <Button
        as='a'
        href='https://github.com/markcnunes/street-photography-hashtags'
        title='View project on GitHub'
        target='_blank'
        rel='noopener noreferrer'
        variant='outlined'
        size='small'
        color='white'
      >
        <svg
          className='github'
          viewBox='0 0 24 24'
          width='24'
          height='24'
          stroke='currentColor'
          strokeWidth='2'
          fill='none'
          strokeLinecap='round'
          strokeLinejoin='round'
        >
          <path d='M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22'></path>{' '}
        </svg>
        GitHub
      </Button>
    </StyledHeader>
  );
};

export default Header;
