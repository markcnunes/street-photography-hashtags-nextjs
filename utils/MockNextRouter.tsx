import { RouterContext } from 'next/dist/next-server/lib/router-context';
import { NextRouter } from 'next/router';
import { ReactNode } from 'react';

interface MockNextRouterProps extends Partial<NextRouter> {
  children?: ReactNode;
}

/**
 * Override the default NextRouter render with different attributes.
 * e.g. change the path or locale for tests.
 */

const MockNextRouter = ({ children, ...props }: MockNextRouterProps) => {
  const mockRouter: NextRouter = {
    route: '/',
    pathname: '',
    query: {},
    asPath: '/',
    basePath: '',
    locale: 'en',
    locales: [''],
    defaultLocale: 'en',
    // domainLocales?: DomainLocales,
    isLocaleDomain: true,
    push: jest.fn(() => Promise.resolve(true)),
    replace: jest.fn(() => Promise.resolve(true)),
    reload: jest.fn(() => Promise.resolve(true)),
    back: jest.fn(() => Promise.resolve(true)),
    prefetch: jest.fn(),
    beforePopState: jest.fn(() => Promise.resolve(true)),
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
    isFallback: false,
    isReady: true,
    isPreview: false,
  };

  return (
    <RouterContext.Provider value={{ ...mockRouter, ...props }}>
      {children}
    </RouterContext.Provider>
  );
};

export default MockNextRouter;
