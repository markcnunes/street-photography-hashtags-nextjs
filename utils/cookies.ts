// import ReactGA from 'react-ga';

import GA4React from 'ga-4-react';

export const cookieConsent = 'sphCookieconsent';

/*
 * Check if cookie consentment is true before initializing and rendering it
 */
export const getCookie = (key: string): string | undefined => {
  var b = document.cookie.match('(^|;)\\s*' + key + '\\s*=\\s*([^;]+)');
  return b ? b.pop() : '';
};

export const gaCode = 'G-W7QKGLQ9YV';
const initializeCookies = () => {
  if (!window.location.href.includes('localhost')) {
    const ga4react = new GA4React(gaCode);
    ga4react.initialize();
    // reload page to get the track running
    window.location.reload();
  }
};

export const renderCookies = (): any => {
  getCookie(cookieConsent) === 'true' && initializeCookies();
};

export default initializeCookies;
