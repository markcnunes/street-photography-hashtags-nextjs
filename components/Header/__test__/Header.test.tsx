import React from 'react';
import MockNextRouter from 'utils/MockNextRouter';
import { render } from 'utils/test-utils';

import Header from '../Header';

describe('<Header />', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <MockNextRouter>
        <Header />
      </MockNextRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
