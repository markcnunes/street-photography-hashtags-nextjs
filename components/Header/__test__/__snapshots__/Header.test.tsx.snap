// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 20px;
  color: #fff;
}

.emotion-0 .logo {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 10px;
}

.emotion-0 .logo img {
  width: 100px;
}

.emotion-0 .github {
  margin-right: 5px;
  width: 20px;
  height: 20px;
}

@media (max-width: 400px) {
  .emotion-0 {
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
  }
}

@media (max-width: 780px) {
  .emotion-0 nav {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
    width: 100%;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
  }
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  border: 0;
  outline: 0;
  cursor: pointer;
  border-radius: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  padding: 10px;
  border: 1px solid transparent;
  border-radius: 4px;
}

.emotion-2 a.is-active {
  border-color: #536075;
}

.emotion-2 a:hover {
  border-color: currentColor;
}

.emotion-2 a+a {
  margin-left: 10px;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: auto;
  -ms-flex-pack: auto;
  -webkit-justify-content: auto;
  justify-content: auto;
  border: 0;
  outline: 0;
  cursor: pointer;
  width: auto;
  opacity: 1;
  border-radius: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
  color: #1c1c27;
  background: #fff;
  font-size: 14px;
  padding: 8px 16px;
  border-radius: 4px;
  border: 1px solid currentColor;
  background: transparent;
  color: #fff;
}

.emotion-4:disabled {
  cursor: auto;
}

.emotion-4:hover {
  background: #fff;
}

.emotion-4:disabled {
  color: rgba(83,96,117,0.5);
}

.emotion-4:hover {
  background: rgba(255,255,255,0.05);
}

.emotion-4:hover:disabled {
  background: transparent;
  color: rgba(83,96,117,0.5);
}

<header
    class="emotion-0 emotion-1"
  >
    <a
      class="logo"
      href="/"
      title="home"
    >
      <img
        alt="Logo"
        src="./logo-white.svg"
      />
    </a>
    <nav
      aria-label="main navigation"
      class="emotion-2 emotion-3"
      role="navigation"
    >
      <a
        class="is-active"
        href="/"
      >
        Categories
      </a>
      <a
        class="navbar-item"
        href="/about"
      >
        About
      </a>
      <a
        class="navbar-item"
        href="/contribution"
      >
        Contribution
      </a>
    </nav>
    <a
      class="emotion-4 emotion-5"
      color="white"
      href="https://github.com/markcnunes/street-photography-hashtags"
      rel="noopener noreferrer"
      target="_blank"
      title="View project on GitHub"
    >
      <svg
        class="github"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
        />
         
      </svg>
      GitHub
    </a>
  </header>
</DocumentFragment>
`;
